### DISCUSSION TITLE
Gitlab merge request without API token

### DISCUSSION THREAD

**Discussion by sualko** [WeblateOrg/weblate#13058] [Ideas]
We have a dedicated Weblate server and tried to use the Gitlab merge request feature. Since nothing is documented for the required scope of the API token we just used repo read/write. This seams not enough and now we have to contact support again, because Gitlab does not allow to change the scope afterwards. This process is annoying and therefore I'm wondering why Weblate is not using push options to create the merge request.

https://docs.gitlab.com/ee/topics/git/commit.html#push-options-for-merge-requests

This would allow to create merge requests only with the ssh key.

**Comment by nijel** [comment]
Weblate is not using that because nobody implemented that.

The good starting point would be to create a new feature request here: https://github.com/WeblateOrg/weblate/issues/new?assignees=&labels=&projects=&template=feature_request.yml
